examples/sprite_sheets/combine_images_into_sprite_sheet/combine_images_into_sprite_sheet.lpr

```diff
@@ -69,11 +136,52 @@
       // Lay out image rows from the top:
       Y := OutputImage.Height - (I div Columns + 1) * InputVideo.Height;
       InputVideo.Items[I].DrawTo(OutputImage, X, Y, dmOverwrite);
+      if ReverseY then
+        Y := ((I div Columns + 1) * InputVideo.Height) - InputVideo.Height;
+      if SingleElement then
+        AnimName := ExtractFileName(InputVideo.Items[I].URL)
+      else
+        begin
+          AnimName := 'Index_Not_Found';
+          for K := 0 to Length(ElementList)-1 do
+          begin
+            AnimList := ElementList[K].Split(',');
+            from := strtoint(AnimList[1]);
+            unto := strtoint(AnimList[2]);
+            if StartFromZero then
+              H := I
+            else
+              H := I + 1;
+            if ((H >= from) and (H <= unto)) then
+            begin
+              AnimNameIndex := IntToStr(H);
+              if length(AnimNameIndex) < Padding then
+              begin
+                for A := length(AnimNameIndex) to Padding-1 do
+                  AnimNameIndex := '0' + AnimNameIndex;
+              end;
+              AnimName := AnimList[0] + '_' + AnimNameIndex + FileExt;
+              break;
+            end;
+          end;
+        end;
+        XMLFileList.add('<SubTexture name="' + AnimName +
+                      '" x="' + X.ToString +
+                      '" y="' + Y.ToString +
+                      '" width="' + InputVideo.Items[I].Width.ToString +
+                      '" height="' + InputVideo.Items[I].Height.ToString + '"/>');
     end;
-
+    XMLFileList.Add('</TextureAtlas>');
+    XMLFileList.SaveToFile(ExtractFilePath(OutputUrl) + ChangeFileExt(XMLFileName, '') + '.xml');
     SaveImage(OutputImage, OutputUrl);
```

Comment:
As above  -- do not asssume that URLs are just filenames.

So `ChangeFileExt` should be `ChangeURIExt`.

And combining paths (absolute + relative) can be done by `CombineURI(...)` (from `CastleURIUtils` unit).